### PR TITLE
Add and improve documentation for absorption corrections 

### LINKED ISSUE #16044: Implement SetSample algorithm
This new algorithm is described in the design document for the generic-shape [absorption corrections](https://github.com/mantidproject/documents/blob/master/Design/AbsorptionCorrections/GenericShapeAbsorptionCorrection.md). It will form part of the workflow for the corrections procedure.

### PR DESCRIPTION
**Pull request by martyngigg**
**To test:**

Review the documentation changes to ensure they make sense.

I had to make a small change to `SetSample` so that the `Axis` argument was interpreted as an integer and not a double. I only picked this up when writing the usage tests.

Fixes #16044 

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/algorithms/MonteCarloAbsorption-v1.rst [hunk 1]
```diff
@@ -9,84 +9,145 @@
...
-.. Ticket 9644 is in place to improve the python exports and expand this example
+References
+----------
 
-.. code-block:: python
-    
-   # The algorithm does allow you to set a complex sample environment
-   # of different materials and shapes, but some of the required methods
-   # are not exported to python yet.  This will come.
+.. [#DAR] Darwin, C. G., *Philos. Mag.*, **43** 800 (1922)
```
Comment (peterfpeterson) [inline docs/source/algorithms/MonteCarloAbsorption-v1.rst]: Add DOIs similar to SetSampleMaterial

**Comment by peterfpeterson** [comment]
Other than adding the DOI links, this looks good to me.

**Comment by martyngigg** [comment]
DOI links added.

**Comment by martyngigg** [comment]
The documentation build has passed now.

**Comment by peterfpeterson** [comment]
LGTM :shipit:

**Pull request merged**